interfaces/repo.go

```diff
@@ -34,23 +34,23 @@ type Repository struct {
 	m      sync.Mutex
 	ifaces map[string]Interface
 	// Indexed by [snapName][plugName]
-	plugs map[string]map[string]*Plug
-	slots map[string]map[string]*Slot
+	plugs map[string]map[string]*snap.PlugInfo
+	slots map[string]map[string]*snap.SlotInfo
 	// given a slot and a plug, are they connected?
-	slotPlugs map[*Slot]map[*Plug]*Connection
+	slotPlugs map[*snap.SlotInfo]map[*snap.PlugInfo]*Connection
 	// given a plug and a slot, are they connected?
-	plugSlots map[*Plug]map[*Slot]*Connection
+	plugSlots map[*snap.PlugInfo]map[*snap.SlotInfo]*Connection
```

Comment:
Same here.

Reply:
Thaks, indeed, this sounds like an idea. I'd like to address this concern in one of the followup PRs, I think this internal detail is not critical yet.